src/context.rs

```diff
@@ -1 +1,33 @@
-pub struct Context {}
+use crate::event::Event;
+use crate::AResult;
+use chrono::{DateTime, Utc};
+use serde::Serialize;
+use std::thread;
+use std::cell::RefCell;
+
+#[derive(Debug, Clone)]
+pub struct Context {
+    pub key: String,
+    pub context_id: String,
```

Comment:
Whats the difference between key and context_id? Are they absolutely necessary?

Reply:
Key is the unique identifier, and context_id can be same for multiple calls.